plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/button/frontend.ts

```diff
@@ -51,14 +55,59 @@ const { state: addToCartWithOptionsState } = store< AddToCartWithOptionsStore >(
 	{ lock: universalLock }
 );
 
+const getMatchedCartItem = (
+	cartItem: OptimisticCartItem,
+	selectedItem: CartVariationItem[]
+) => {
+	if (
+		! Array.isArray( cartItem.variation ) ||
+		! Array.isArray( selectedItem )
+	)
+		return false;
+	if ( cartItem.variation.length !== selectedItem.length ) return false;
+
+	return cartItem.variation.every(
+		( {
+			raw_attribute,
+			value,
+		}: {
+			raw_attribute: string;
+			value: string;
+		} ) =>
+			selectedItem.some( ( item ) => {
+				const selectedItemAttr = item.attribute;
+				const selectedItemValue = item.value.toLowerCase();
+
+				return (
+					selectedItemAttr === raw_attribute &&
+					( selectedItemValue === value.toLowerCase() ||
+						( item.value && value === '' ) )
+				);
+			} )
+	);
+};
+
 const productButtonStore = {
 	state: {
 		get quantity(): number {
-			const product = wooState.cart?.items.find(
-				( item ) => item.id === state.productId
+			const selectedAttribute =
+				addToCartWithOptionsState?.selectedAttributes;
```

Comment:
I would move this directly before the line where `selectedAttribute` is used. Right before:

```JS
const selectedProduct = product.find( ( item ) =>
	getMatchedCartItem( item, selectedAttribute )
);
```